mock/mock.go

```diff
@@ -766,7 +766,25 @@ const (
 // AnythingOfTypeArgument contains the type of an argument
 // for use when type checking.  Used in Diff and Assert.
 //
-// Deprecated: this is an implementation detail that must not be used. Use [AnythingOfType] instead.
+// Deprecated: this is an implementation detail that must not be used. Use the [AnythingOfType] constructor instead.
+//
+// If AnythingOfTypeArgument is used as a function return value type you can usually replace
+// the function with a variable. Example:
+//
+//	func anyString() mock.AnythingOfTypeArgument {
+//	    return mock.AnythingOfType("string")
+//	}
+//
+// Can be replaced by:
+//
+//	func anyString() interface{} {
+//	    return mock.IsType("")
+//	}
+//
+// It could even be replaced with a variable:
+//
+//	var anyString = mock.AnythingOfType("string")
+//	var anyString = mock.IsType("") // alternative
```

Comment:
This suggests to replace with a variable, then shows an example which does not do that. Then it suggests again that it can be replaced with a variable and shows two examples.

This also suggests IsType as an alternative, which I agree is a better matcher, but it's not what the deprecation line says to use. It might also not be as easy to switch to for people with tabled tests that already define the type by its string name.

I rendered this with pkgsite and with the amount of code blocks it becomes hard to read, especially with the "bad" code examples present, users might take them as a "good" example.

How about something more simple?

```suggestion
// Deprecated: this is an implementation detail that must not be used. Use the [AnythingOfType] constructor instead, example:
//
//	m.On("Do", mock.AnythingOfType("string"))
//
// All explicit type declarations can be replaced with interface{} as is expected by [Mock.On], example:
//
//	func anyString interface{} {
//		return mock.AnythingOfType("string")
//	}
```

Which will look like:
<img width="1316" alt="image" src="https://github.com/stretchr/testify/assets/1909753/2418f0c9-4377-4361-b309-2940611696bf">